src/utils/HTTPTransport/HTTPTransport.ts

```diff
@@ -0,0 +1,88 @@
+enum METHOD {
+    GET = 'GET',
+    POST = 'POST',
+    PUT = 'PUT',
+    PATCH = 'PATCH',
+    DELETE = 'DELETE',
+}
+
+type Options = {
+    method?: METHOD;
+    timeout?: number;
+    headers?: Record<string, string>;
+    data?: any;
+};
+
+type OptionsWithoutMethod = Omit<Options, 'method'>;
+
+type HTTPMethod = (url: string, options?: OptionsWithoutMethod) => Promise<unknown>
```

Comment:
Можно сделать функцию дженериком, чтобы иметь возможность описать тип возвращаемого значения извне
```suggestion
type HTTPMethod = <Res>(url: string, options?: OptionsWithoutMethod) => Promise<Res>
```

Reply:
done